integration_test.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+func TestFetchWithBranchOption(t *testing.T) {
+	t.Parallel()
+
+	tmpDownloadPath := createTempDir(t, "fetch-branch-test")
+
+	cases := []struct {
+		name       string
+		repoUrl    string
+		branchName string
+		sourcePath string
+	}{
+		// Test on a public repo whose sole purpose is to be a test fixture for this tool
+		{"branch option with public repo", "https://github.com/gruntwork-io/fetch-test-public", "sample-branch", "/"},
+
+		// Private repo equivalent
+		{"branch option with private repo", "https://github.com/gruntwork-io/fetch-test-private", "sample-branch", "/"},
+	}
+
+	for _, tc := range cases {
+		tc := tc
```

Comment:
NIT: add our usual comment block.

```go
    // The following is necessary to make sure tc's values don't
    // get updated due to concurrency within the scope of t.Run(..) below
```

Reply:
@brikis98 see my comment above. Should I skip this comment as I don't think they'll run in parallel?

Reply:
done.